polkadot/node/subsystem-bench/src/core/mock/network_bridge.rs

```diff
@@ -0,0 +1,280 @@
+					_ => unimplemented!("Peer recipient not supported yet"),
+				};
+				// Account our sent request bytes.
+				self.network.peer_stats(0).inc_sent(outgoing_request.payload.encoded_size());
+				// Account for remote received request bytes.
+				self.network
+					.peer_stats_by_id(authority_discovery_id.clone())
+					.inc_received(outgoing_request.payload.encoded_size());
+
+				let validator_index: usize = outgoing_request.payload.index.0 as usize;
+				let candidate_hash = outgoing_request.payload.candidate_hash;
+
+				let candidate_index = self
+					.availabilty
+					.candidate_hashes
+					.get(&candidate_hash)
+					.expect("candidate was generated previously; qed");
+				gum::warn!(target: LOG_TARGET, ?candidate_hash, candidate_index, "Candidate mapped to index");
+
+				let chunk: ChunkResponse =
+					self.availabilty.chunks.get(*candidate_index as usize).unwrap()
+						[validator_index]
+						.clone()
+						.into();
+				let mut size = chunk.encoded_size();
+
+				let response = if random_error(self.config.error) {
+					// Error will not account to any bandwidth used.
+					size = 0;
+					Err(RequestFailure::Network(OutboundFailure::ConnectionClosed))
+				} else {
+					Ok(req_res::v1::ChunkFetchingResponse::from(Some(chunk)).encode())
+				};
+
+				let authority_discovery_id_clone = authority_discovery_id.clone();
+
+				let future = async move {
+					let _ = outgoing_request.pending_response.send(response);
+				}
+				.boxed();
+
+				let future_wrapper = async move {
+					// Forward the response to the ingress channel of our node.
+					// On receive side we apply our node receiving rate limit.
+					let action =
+						NetworkAction::new(authority_discovery_id_clone, future, size, None);
+					ingress_tx.send(action).unwrap();
+				}
+				.boxed();
+
+				NetworkAction::new(
+					authority_discovery_id,
+					future_wrapper,
+					size,
+					// Generate a random latency based on configuration.
+					random_latency(self.config.latency.as_ref()),
+				)
+			},
+			Requests::AvailableDataFetchingV1(outgoing_request) => {
+				let candidate_hash = outgoing_request.payload.candidate_hash;
+				let candidate_index = self
+					.availabilty
+					.candidate_hashes
+					.get(&candidate_hash)
+					.expect("candidate was generated previously; qed");
+				gum::debug!(target: LOG_TARGET, ?candidate_hash, candidate_index, "Candidate mapped to index");
+
+				let authority_discovery_id = match outgoing_request.peer {
+					req_res::Recipient::Authority(authority_discovery_id) => authority_discovery_id,
+					_ => unimplemented!("Peer recipient not supported yet"),
+				};
+				// Account our sent request bytes.
+				self.network.peer_stats(0).inc_sent(outgoing_request.payload.encoded_size());
+				// Account for remote received request bytes.
+				self.network
+					.peer_stats_by_id(authority_discovery_id.clone())
+					.inc_received(outgoing_request.payload.encoded_size());
```

Comment:
Any new request needs to implement some behaviour and that is fine. Accounting here requires parsing the message since it's not really possible to know the peer recipient without doing so.

Reply:
I will rework the network and stats in a followup PR. The new implementation should be more generic and accounting will be done in a single place.